File: WatchOS/DailyWatch/DailyWatch Watch App/TemporalNavHeader.swift
import SwiftUI
import WatchKit

struct TemporalNavHeader: View {
    let title: String
    let canGoForward: Bool
    let accentColor: Color
    let onPrevious: () -> Void
    let onNext: () -> Void
    
    var body: some View {
        HStack(spacing: 6) {
            Button(action: {
                WKInterfaceDevice.current().play(.click)
                onPrevious()
            }) {
                Image(systemName: "chevron.left")
                    .font(.system(size: 13, weight: .bold))
                    .foregroundColor(accentColor)
                    .frame(width: 28, height: 26)
                    .background(Color.white.opacity(0.12))
                    .cornerRadius(8)
            }
            .buttonStyle(.plain)
            
            Text(title)
                .font(.system(size: 13, weight: .semibold))
                .foregroundColor(.primary)
                .lineLimit(1)
                .minimumScaleFactor(0.8)
                .frame(maxWidth: .infinity, alignment: .center)
            
            if canGoForward {
                Button(action: {
                    WKInterfaceDevice.current().play(.click)
                    onNext()
                }) {
                    Image(systemName: "chevron.right")
                        .font(.system(size: 13, weight: .bold))
                        .foregroundColor(accentColor)
                        .frame(width: 28, height: 26)
                        .background(Color.white.opacity(0.12))
                        .cornerRadius(8)
                }
                .buttonStyle(.plain)
            } else {
                // Invisible placeholder to keep the center title perfectly centered
                Color.clear
                    .frame(width: 28, height: 26)
            }
        }
        .padding(.horizontal, 4)
        .padding(.vertical, 2)
    }
}

File: WearOS/app/src/main/java/com/intellidream/daily/wearos/presentation/components/TemporalNavHeader.kt
package com.intellidream.daily.wearos.presentation.components

import android.view.HapticFeedbackConstants
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalView
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.wear.compose.material.Text

@Composable
fun TemporalNavHeader(
    title: String,
    canGoForward: Boolean,
    accentColor: Color,
    onPrevious: () -> Unit,
    onNext: () -> Unit,
    modifier: Modifier = Modifier
) {
    val view = LocalView.current

    Row(
        modifier = modifier
            .fillMaxWidth()
            .height(32.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        // Previous Button (‹)
        Box(
            modifier = Modifier
                .size(width = 36.dp, height = 28.dp)
                .clip(RoundedCornerShape(8.dp))
                .background(Color.White.copy(alpha = 0.12f))
                .clickable {
                    view.performHapticFeedback(HapticFeedbackConstants.KEYBOARD_TAP)
                    onPrevious()
                },
            contentAlignment = Alignment.Center
        ) {
            Text(
                text = "‹",
                color = accentColor,
                fontSize = 17.sp,
                fontWeight = FontWeight.Bold
            )
        }

        // Centered Date / Week Label
        Text(
            text = title,
            color = Color.White,
            fontSize = 12.sp,
            fontWeight = FontWeight.Medium,
            textAlign = TextAlign.Center,
            modifier = Modifier.weight(1f)
        )

        // Next Button (›) - hidden when on current date/week
        if (canGoForward) {
            Box(
                modifier = Modifier
                    .size(width = 36.dp, height = 28.dp)
                    .clip(RoundedCornerShape(8.dp))
                    .background(Color.White.copy(alpha = 0.12f))
                    .clickable {
                        view.performHapticFeedback(HapticFeedbackConstants.KEYBOARD_TAP)
                        onNext()
                    },
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = "›",
                    color = accentColor,
                    fontSize = 17.sp,
                    fontWeight = FontWeight.Bold
                )
            }
        } else {
            // Invisible placeholder for balanced layout centering
            Box(modifier = Modifier.size(width = 36.dp, height = 28.dp))
        }
    }
}
